src/model/Model.lua

```diff
@@ -168,7 +168,32 @@ end
      * @return the model text.
 ]]
 function Model:saveModelToText()
-
+	local tokenPatterns={}
+	for _,ptype in pairs({"r","p"}) do
+		for _,token in pairs(self.model[ptype][ptype].tokens) do
+			tokenPatterns[token]=string.gsub (string.gsub (token,"^p_","p."),"^r_","r.")
+		end
+	end
+	local s=""
+	local writeString=function(sec)
+		for ptype,_ in pairs(self.model[sec]) do
+			value=self.model[sec][ptype].value
+			for tokenPattern,newToken in pairs(tokenPatterns) do
+				value=string.gsub(value,tokenPattern,newToken)
+			end
+			s=s..sec.."="..value.."\n"
```

Comment:
This wouldn't work, instead create a new local variable at the function start and use that instead of `s`, also return it before the function end.